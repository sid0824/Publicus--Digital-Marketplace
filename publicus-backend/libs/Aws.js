import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});