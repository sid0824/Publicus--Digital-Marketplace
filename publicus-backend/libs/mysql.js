import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};