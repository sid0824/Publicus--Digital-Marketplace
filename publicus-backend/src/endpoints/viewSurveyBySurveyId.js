import {runQuery} from "../../libs/mysql";
import {failure, json_success} from "../util/ResponseUtil";

//Generates SQL Query (returns surveys with a matching ID)
const selectSurveyQuery = (surveyId) => {
  return `SELECT *
          FROM Surveys
          WHERE surveyId = '${surveyId}'`;
};

//Generates SQL Query (Returns a list of tags that a surveyId has)
const selectTagQuery = (surveyId) => {
  return `SELECT tag 
          FROM Tags
          WHERE surveyId = '${surveyId}'`;
};

export async function main(event, context) {
  try {
    const results = await runQuery(selectSurveyQuery(event.pathParameters.surveyId));
    if (!results.length) {
      return failure();
    }
    const form = results[0];

    //Modify survey result to also include a list of tags for that survey
    const tagSql = selectTagQuery(form.surveyId);
    const tags = await runQuery(tagSql);

    return json_success({
      ...form,
      tags
    });
  } catch (err) {
    console.error(err);
    return failure();
  }
}
