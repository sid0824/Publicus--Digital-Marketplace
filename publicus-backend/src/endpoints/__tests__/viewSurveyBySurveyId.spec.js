import { main } from '../viewSurveyBySurveyId';
import { runQuery } from "../../../libs/mysql";

jest.mock("../../../libs/mysql");

const mockEvent = {
  requestContext: {
    identity: {
      cognitoIdentityId: 'MOCK_COGNITO_ID'
    },
  },
  body: JSON.stringify({
  }),
  pathParameters: {
    surveyId: 'MOCK_SURVEY_ID'
  }
};

describe('viewSurveyBySurveyId', () => {

  beforeEach(() => {
  });

  afterEach(() => {
    runQuery.mockReset();
  });

  test(`should respond with failure when an error occurs`, async () => {
    runQuery.mockImplementation(() => {
      throw new Error('ERROR')
    });
    const response = await main(mockEvent);
    expect(response).toMatchSnapshot();
  });

  test(`should return the survey of the passed id`, async () => {
    runQuery
      .mockResolvedValueOnce([{ surveyId: 'SURVEY_ID' }])
      .mockResolvedValueOnce([{ tag: 'TAG' }]);
    const response = await main(mockEvent);
    expect(runQuery.mock.calls[0][0]).toMatchSnapshot();
    expect(runQuery.mock.calls[1][0]).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  });

  test(`should fail if survey ID doesnt exist`, async () => {
    runQuery
      .mockResolvedValueOnce([]);
    const response = await main(mockEvent);
    expect(response).toMatchSnapshot();
  })

});