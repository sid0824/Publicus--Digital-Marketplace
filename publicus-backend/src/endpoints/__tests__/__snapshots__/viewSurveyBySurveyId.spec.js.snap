// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewSurveyBySurveyId should fail if survey ID doesnt exist 1`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 500,
}
`;

exports[`viewSurveyBySurveyId should respond with failure when an error occurs 1`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 500,
}
`;

exports[`viewSurveyBySurveyId should return the survey of the passed id 1`] = `
"SELECT *
          FROM Surveys
          WHERE surveyId = 'MOCK_SURVEY_ID'"
`;

exports[`viewSurveyBySurveyId should return the survey of the passed id 2`] = `
"SELECT tag 
          FROM Tags
          WHERE surveyId = 'SURVEY_ID'"
`;

exports[`viewSurveyBySurveyId should return the survey of the passed id 3`] = `
Object {
  "body": "{\\"surveyId\\":\\"SURVEY_ID\\",\\"tags\\":[{\\"tag\\":\\"TAG\\"}]}",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "applications/json",
  },
  "statusCode": 200,
}
`;
